onnx/external_data_helper.py

```diff
@@ -112,23 +114,50 @@ def convert_model_to_external_data(model, all_tensors_to_one_file=True, location
               If not specified, will use the model name.
     size_threshold: Threshold for size of data. Only when tensor's data is >= the size_threshold
     it will be converted to external data. To convert every tensor with raw data to external data set size_threshold=0.
+    convert_attribute: If true, convert all tensors to external data
+                       If false, convert only non-attribute tensors to external data
     """
+    tensors = _get_initializer_tensors(model)
+    if convert_attribute:
+        tensors = _get_all_tensors(model)
+
     if all_tensors_to_one_file:
         file_name = Text(uuid.uuid1())
         if location:
             file_name = location
-        for tensor in _get_all_tensors(model):
+        for tensor in tensors:
             if tensor.HasField("raw_data") and sys.getsizeof(tensor.raw_data) >= size_threshold:
                 set_external_data(tensor, file_name)
     else:
-        for tensor in _get_all_tensors(model):
+        for tensor in tensors:
             if tensor.HasField("raw_data") and sys.getsizeof(tensor.raw_data) >= size_threshold:
                 tensor_location = tensor.name
                 if not _is_valid_filename(tensor_location):
                     tensor_location = Text(uuid.uuid1())
                 set_external_data(tensor, tensor_location)
 
 
+def covert_and_save_model_to_external_data(model, model_file_path, all_tensors_to_one_file=True, location=None, size_threshold=1024, convert_attribute=False):
```

Comment:
can we rename this to save_model and add it to onnx/__init__.py along with the original save_model API? So basically we will have 2 save_model APIs ... @gramalingam thoughts?

Reply:
So, remove it from `external_data_helper.py` and add to `init.py`? 
I can do that, but if we rename it to `save_model`, wouldn't there be a conflict since I'm calling the existing `save_model` in the new `save_model` API?

Reply:
I agree, it would be nicer to have them together. One possibility is to have a single API, with extra-parameters determining whether to convert data to external-data.

Reply:
ack, I can add an extra parmeter `convert_data=False` in addition to parameters from the `convert_and_save_model_to_external_data` in the existing `save_model` API in the `init.py` file

We can use the `convert_data` parameter to determine if convert data to external data is needed? 
I will work on this later today, let me know if you have any concerns @askhade @gramalingam thanks!

Reply:
yes makes sense...  1 suggestion rename convert_data to save_as_external_data and set defualt value to False
Something like this: 1 single save_model api : 
save_model(proto, f, format=None, save_as_external_data=False, all_tensors_to_one_file=True, location=None, size_threshold=1024, convert_attributes=False)

Reply:
done, PTAL again when you can!